From bfb76be4076b70aa05db2f5f8ec2c0b0698dd4b0 Mon Sep 17 00:00:00 2001
From: Thierry FOURNIER <tfournier@arpalert.org>
Date: Wed, 4 Nov 2015 10:52:59 +0100
Subject: [PATCH 05/19] BUG/MEDIUM: lua: clean output buffer

When the txn.done() fiunction is called, the ouput buffer is cleaned,
but the associated relative pointer on the HTTP requests elements
is not reseted.

This patch remove this cleanup, because the output buffer may contain
data to forward.
(cherry picked from commit a3308fd8c1851f06b1374da0f3aa005245a88dd3)
---
 src/hlua.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/hlua.c b/src/hlua.c
index 8fc85a6..5ee2337 100644
--- a/src/hlua.c
+++ b/src/hlua.c
@@ -4819,10 +4819,6 @@ __LJMP static int hlua_txn_done(lua_State *L)
 		htxn->s->txn->req.msg_state = HTTP_MSG_CLOSED;
 		htxn->s->txn->rsp.msg_state = HTTP_MSG_DONE;
 
-		/* Trim any possible response */
-		oc->buf->i = 0;
-		htxn->s->txn->rsp.next = htxn->s->txn->rsp.sov = 0;
-
 		/* Note that if we want to support keep-alive, we need
 		 * to bypass the close/shutr_now calls below, but that
 		 * may only be done if the HTTP request was already
-- 
2.4.10

